### PR TITLE
pkgautoversion: only match conforming tags passed via the command line

### PR DESCRIPTION
**Pull request by mjuric**
`bin/pkgautoversion` is the utility that tries to compute a reasonably-looking version from the git information. It can take an argument -- `REF` -- that it may use as a hint when it's constructing a version (e.g., if there are no annotated tags defined at all, then it will construct the version of the form `<REF>-g<SHA1>`).

Before this PR, `pkgautoversion` would check if `REF` was an annotated tag. If it was, it would return `REF` as the version. This worked as expected, as long as one followed a convention to use annotated tags for (numerical) version only (e.g., `10.1`, `10.1.2`, etc.).

The introduction of weekly tags of the form of `w.2015.33`, etc, changed the convention. That lead to many packages having a version such as `w.2015.33`, if the stack was built with `lsstsw` using a command line such as:
```
rebuild -r w.2015.33 -r 2.0.1 lsst_sims
```
(which is what the simulations team is using to generate builds based off a DM weekly).

Such arbitrary versions are generally hard to parse (and order) -- causing problems for `conda` binary packaging and updating. It also leads to rebuilds of packages that have already been built (because though they have the same SHA1, `lsstsw` sees them with a different version). It can also mask the true version of a package -- esp. nasty with 3rd party packages (e.g., `fftw`).

Because of this, this PR changes the behavior to immediately return the tag *only* if it matches the TAG_PATTERN (which is currently defined as `[0-9]*`, but could be tightened further in the future). Otherwise, the version will be computed using the fall-back rules, as if the tag wasn't there.

Here are some before/after versions, to get a feel for the impacts of this change:
```
before: fftw                      8fdba61333025ecee68de6da9c770be41cdb547e  w.2015.33
after:  fftw                      8fdba61333025ecee68de6da9c770be41cdb547e  3.3.3-1-g8fdba61
```
```
before: afw                       f63c96a08c1dbfb65ce2f5fb23e5c986450f32b3  w.2015.33
after:  afw                       f63c96a08c1dbfb65ce2f5fb23e5c986450f32b3  10.1-38-gf63c96a
```

For more see [before](https://s3.amazonaws.com/uploads.hipchat.com/100518/737581/s0HFNB11qbg3NEz/manifest.before.txt) and [after](https://s3.amazonaws.com/uploads.hipchat.com/100518/737581/F0qyJ2I3uy1jgGM/manifest.txt) `lsstsw` manifest files.

W/o this change, it's very difficult to build `conda` packages (and other packaging system that care about the ordering of versions are likely to have similar problems).

### CODE REVIEW

**Comment by mjuric** [comment]
Oh, I also fixed an issue with outdated unit tests. You can run them with:
```
RUN_UNIT=1 pkgautoversion
```

**Comment by frossie** [comment]
Can you please not merge this PR until I can check that it doesn't have unintended consequences on my release process. 

In general I don't expect EUPS behaviour to change from under me so it would be nice to be directly notified of such changes rather than hope to stumble on them in the morass of automated notification messages.

**Comment by mjuric** [comment]
No worries, merging to EUPS' master won't affect the release (I think 1.5.9 is still on ~lsstsw)!

Good point about the DM stack release. Thought this mostly affects sims, you may need this fix for a DM release as well. If you're building from an annotated tag, that may make it pick up something like 'v10.1.rc1' as a version (hasn't happened yet (good!)).

**Comment by frossie** [comment]
@mjuric Okay I've convinced myself you're right :-) The only downside is that letter-derived-tags won't appear in the manifest so people won't be able to quickly see what has drifted (I think some do that with the official tags) but they can live with that.  Actually the system works fine with them as is, but I would like to actually delete the older weekly git tags, so it's better if eups doesn't know anything about them, besides as a published tag name. 

PS. I realise it doesn't affect the current deploy, but it's nice not to have any ticking time bombs anyway, we do update the toolchain occasionally...

**Comment by frossie** [comment]
So in fact it turns out people *can't* live with that. They really do want to see a human readable warm fuzzy string on the manifest list rather than some +gsha. and they are happy to take the rebuild time on the chin to get that. If you can't make this behaviour optional, I suggest you whitelist one character (like o for official) for that usecase. That means you can still ignore things like the weeklies, which is  a good thing.
